modules/gui/task-list.lua

```diff
@@ -57,360 +76,657 @@ local function check_player_permissions(player, task)
-    sprite = 'utility/close_black',
-    tooltip = {'task-list.cancel-tooltip'},
-    style = 'shortcut_bar_button_red'
+	type = "sprite-button",
+	sprite = "utility/collapse_dark",
+	hovered_sprite = "utility/collapse",
+	tooltip = {"task-list.close-tooltip"}
 }
-:style(Styles.sprite22)
-:on_click(function(player, element, _)
-    local task_id = element.parent.name
-    Tasks.set_editing(task_id, player.name)
-end)
-
---- Editing state for a task, contrins a text field and the two edit buttons
--- @element task_editing
-task_editing =
-Gui.element(function(event_trigger, parent, task)
-    local message = task.message
-
-    -- Draw the element
-    local element =
-    parent.add{
-        name = event_trigger,
-        type = 'textfield',
-        text = message,
-        clear_and_focus_on_right_click = true
+:style(Styles.sprite22):on_click(
+    function(player, _, _)
+        PlayerSelected:set(player, nil)
+    end
+)
+
+--- Button to delete the task
+-- @element view_task_delete_button
+local view_task_delete_button =
+    Gui.element {
+    type = "button",
+    caption = {"", "[img=utility/trash] ", {"task-list.delete"}},
+    tooltip = {"task-list.delete-tooltip"},
+    style = "shortcut_bar_button_red"
+}:style(Styles.footer_button):on_click(
+    function(player, _, _)
+        local selected = PlayerSelected:get(player)
+        PlayerSelected:set(player, nil)
+        Tasks.remove_task(selected)
+    end
+)
+
+--- Subfooter inside the tasklist container
+-- @element task_view_footer
+local task_view_footer =
+    Gui.element(
+    function(_, parent)
+        local footer = subfooter_frame(parent, "view")
+        local flow = footer.add{ type = "flow" }
+        subfooter_label(flow, {"task-list.view-footer-header"})
+        local alignment = Gui.alignment(flow)
+        view_task_close_button(alignment)
+        local title_label =
+            footer.add {
+            type = "label",
+            name = "title"
+        }
+        title_label.style.padding = 4
+        title_label.style.font = "default-bold"
+        title_label.style.single_line = false
+        local body_label =
+            footer.add {
+            type = "label",
+            name = "body"
+        }
+        body_label.style.padding = 4
+        body_label.style.single_line = false
+
+        local action_flow = subfooter_actions(footer)
+        view_task_delete_button(action_flow)
+        view_task_edit_button(action_flow)
+        return footer
+    end
+)
+
+local message_pattern = "(.-)\n(.*)"
+
+--- Parce a string into a message object with title and body
+-- @tparam string str message data
+local function parse_message(str)
+    -- Trimm the spaces of the string
+    local trimmed = string.gsub(str, "^%s*(.-)%s*$", "%1")
+    local message = { title = "", body = "" }
+    local title, body = string.match(trimmed, message_pattern)
+    if not title then
+        -- If it doesn't match the patter return the str as a title
```

Comment:
```suggestion
        -- If it doesn't match the pattern return the str as a title
```
str is one of the arguments, which specific string is this referring to (at least to me str is often used as a generic type for strings)